src/goose/toolkit/developer.py

```diff
@@ -114,21 +120,20 @@ def patch_file(self, path: str, before: str, after: str) -> str:
         if content.count(before) < 1:
             raise ValueError("The before content was not found in file, be careful that you recreate it exactly.")
 
-        content = content.replace(before, after)
-        _path.write_text(content)
-
-        output = f"""
-```{language}
-{before}
-```
-->
-```{language}
-{after}
-```
-"""
         self.notifier.log(Rule(RULEPREFIX + path, style=RULESTYLE, align="left"))
-        self.notifier.log(Markdown(output))
-        return "Succesfully replaced before with after."
+        show_diff(path, before, after)
+        to_path_file = True
+        if is_in_safe_mode():
+            self.notifier.stop()
+            to_path_file = confirm("Would like to continue to make change?")
+            self.notifier.start()
+        if to_path_file:
+            self.notifier.status(f"editing {path}")
+            content = content.replace(before, after)
+            _path.write_text(content)
+            return "Successfully replaced before with after."
+        else:
+            return {"result": "User chooses not to make the change on this file. skip the change", "task_status": "skipped"}
```

Comment:
:nit: i like active voice for this `User rejected suggested changes to {file}. Skipping edits.`